databend-jdbc/src/main/java/com/databend/jdbc/DatabendClientLoadBalancingPolicy.java

```diff
@@ -0,0 +1,118 @@
+package com.databend.jdbc;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Random;
+
+public class DatabendClientLoadBalancingPolicy {
+    static class DisabledPolicy extends DatabendClientLoadBalancingPolicy {
+        @Override
+        public String toString() {
+            return "Disabled";
+        }
+        // do nothing
+    }
+    static class RandomPolicy extends DatabendClientLoadBalancingPolicy {
+        @Override
+        protected URI pickUri(String query_id, DatabendNodes nodes) {
+            List<URI> uris = nodes.getUris();
+
+            if (uris == null || uris.isEmpty()) {
+                return null;
+            }
+
+            // Generate a deterministic integer based on the query_id
+            int deterministicValue = getQueryHash(query_id);
+
+            // Use the deterministic value to select a URI
+            int index = Math.abs(deterministicValue) % uris.size();
+
+            return uris.get(index);
+        }
+
+        @Override
+        public String toString() {
+            return "Random";
+        }
+    }
+
+    static class RoundRobinPolicy extends DatabendClientLoadBalancingPolicy {
+        @Override
+        protected URI pickUri(String query_id, DatabendNodes nodes) {
+            List<URI> uris = nodes.getUris();
+
+            if (uris == null || uris.isEmpty()) {
+                return null;
+            }
+
+            // Use round robin to select a URI
+            int index = nodes.index.getAndUpdate(v -> v + 1 >= uris.size() ? 0 : v + 1);
+
+            return uris.get(index);
+        }
+
+        @Override
+        public String toString() {
+            return "RoundRobin";
+        }
+    }
+    /**
+     * Policy that disable load balance and always use the first node.
+     */
+    public static final String DISABLED = "disabled";
```

Comment:
Maybe `DISABLE`, `RANDOM`, `ROUND_ROBIN` use ENUM is better ?

Reply:
I think for now we could keep it simple.